### PR TITLE
fix(app): move queries to query package

### PR DESCRIPTION
**Pull request by adlerhurst**

### CODE REVIEW

File: internal/auth/repository/eventsourcing/eventstore/auth_request.go [hunk 1]
```diff
@@ -117,18 +118,23 @@ func (repo *AuthRequestRepo) CreateAuthRequest(ctx context.Context, request *dom
 		return nil, err
 	}
 	request.ID = reqID
-	app, err := repo.View.ApplicationByClientID(ctx, request.ApplicationID)
+	//DISCUSS: the 3 following calls refer to the query side but we are on the command side
```
Comment (livio-a) [inline internal/auth/repository/eventsourcing/eventstore/auth_request.go:121]: 👍  we should add a todo for auth request refactoring / login improvements
Spontaneously I'd suggest moving it to the oidc pkg in the api layer

File: internal/api/oidc/client.go [hunk 2]
```diff
@@ -125,11 +124,11 @@ func (o *OPStorage) SetUserinfoFromToken(ctx context.Context, userInfo oidc.User
 		return errors.ThrowPermissionDenied(nil, "OIDC-Dsfb2", "token is not valid or has expired")
 	}
 	if token.ApplicationID != "" {
-		app, err := o.repo.ApplicationByClientID(ctx, token.ApplicationID)
+		app, err := o.query.AppByID(ctx, token.ApplicationID)
```
Comment (livio-a) [inline internal/api/oidc/client.go:127]:
```suggestion
		app, err := o.query.AppByOIDCClientID(ctx, token.ApplicationID)
```

Comment (livio-a) [reply internal/api/oidc/client.go:127]: in token view there's a naming / using error where the client_id is stored in `ApplicationID`

we'll have to fix this with the new token projection:
either rename it or change the "content"

**Review by livio-a — CHANGES_REQUESTED** [review]
some small changes and we might discuss using an interface for app and the config types

File: internal/auth/repository/eventsourcing/eventstore/auth_request.go (hunk 1 above)
Comment (adlerhurst) [reply internal/auth/repository/eventsourcing/eventstore/auth_request.go]: added a TODO in `internal/api/oidc/auth_request.go`

File: internal/api/oidc/client.go (hunk 2 above)
Comment (adlerhurst) [reply internal/api/oidc/client.go]: do we have a tracking issue for that? I received a 418 from https://livios.brain/caos/zitadel/issues?name=auth%20request%20refactoring  🥴

(added a comment on the tracking issue of queries)

File: internal/auth/repository/eventsourcing/eventstore/auth_request.go (hunk 1 above)
Comment (adlerhurst) [reply internal/auth/repository/eventsourcing/eventstore/auth_request.go]: do we have a tracking issue for that refactoring? I received a 418 from https://livios.brain/caos/zitadel/issues?name=auth%20request%20refactoring  🥴

**Review by adlerhurst — COMMENTED** [review]
@livio-a we have to discuss the TODO's

File: internal/domain/application_oidc.go [hunk 3]
```diff
@@ -181,16 +181,35 @@ func GetOIDCCompliance(version OIDCVersion, appType OIDCApplicationType, grantTy
 	case OIDCVersionV1:
 		return GetOIDCV1Compliance(appType, grantTypes, authMethod, redirectUris)
 	}
-	return nil
+	return nil //TODO: shouldn't we add unsuported version problem?
```
Comment (livio-a) [inline internal/domain/application_oidc.go:184]: i agree... should never happen, but if an error is much better

File: internal/domain/application_oidc_test.go [hunk 4]
```diff
@@ -185,3 +187,489 @@ func TestApplicationValid(t *testing.T) {
...
+		},
+		{
+			name: "not allowed combination",
+			args: args{
+				appType:      OIDCApplicationTypeNative,
+				redirectUris: []string{"https://nirvana.com", "cutom://nirvana.com"},
+			},
+			want: &Compliance{
+				NoneCompliant: false, //TODO: imo this should be true
+				Problems:      []string{"Application.OIDC.V1.NotAllCombinationsAreAllowed"},
```
Comment (livio-a) [inline internal/domain/application_oidc_test.go:438]: IMO https is always allowed and custom for native is ok as well

the way we check at the moment we know that it's not "not compliant" but we're not 100% sure if it is compliant in all cases... we might consider improving the checks (in the future?)

File: internal/domain/application_oidc_test.go [hunk 5]
```diff
@@ -185,3 +187,489 @@ func TestApplicationValid(t *testing.T) {
...
+			want: &Compliance{
+				NoneCompliant: true,
+				Problems:      []string{"Application.OIDC.V1.Code.RedirectUris.CustomOnlyForNative"},
+			},
+		},
+		{
+			name: "only http protocol, app type user agent",
+			args: args{
+				redirectUris: []string{"http://nirvana.com"},
+				appType:      OIDCApplicationTypeUserAgent, //TODO: shouldnt it be type web?
```
Comment (livio-a) [inline internal/domain/application_oidc_test.go:561]: with web http would be allowed

**Review by livio-a — APPROVED** [review]

**Pull request merged**